invenio_rdm_records/requests/community_submission.py

```diff
@@ -84,11 +87,11 @@ class DeclineAction(RequestAction):
 
     def execute(self, identity, uow):
         """Execute action."""
-        # TODO: The requestor/creator should still be able to see the request
-        # from the upload form (so we likely cannot remove the request id from
-        # the record). Probably the user has to manually delete/update the
-        # the review? I.e. how does the user submit the record to a new
-        # community.
+        # Unset review from record (still accessible from request)
+        # This means that the receiver (curator) won't have access anymore
+        # The creator (uploader) should still have access to the record/draft
+        draft = self.request.topic.resolve()
+        draft.parent.review = None
```

Comment:
of course, this is gonna be an issue w.r.t. status display on the deposit page/upload form
however, as discussed yesterday, this should probably be taken care of in the (near) future